src/www/ui/api/Controllers/UploadController.php

```diff
@@ -650,4 +651,73 @@ private function areAgentsScheduled($uploadId, $agents, $response)
     }
     return true;
   }
+
+  /**
+   * Set permissions for a upload in a folder for different groups
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function setUploadPermissions($request, $response, $args)
+  {
+    $returnVal = null;
+    // checking if the scheduler is running or not
+    $commu_status = fo_communicate_with_scheduler('status', $response_from_scheduler, $error_info);
+    if ($commu_status) {
+      // Initialising upload-permissions plugin
+      global $container;
+      $restHelper = $container->get('helper.restHelper');
+      $uploadPermissionObj = $restHelper->getPlugin('upload_permissions');
+
+      $dbManager = $this->dbHelper->getDbManager();
+      // parsing the request body
+      $reqBody = $this->getParsedBody($request);
+
+      $folder_pk = intval($reqBody['folderId']);
+      $upload_pk = intval($args['id']);
+      $upload = $this->uploadAccessible($this->restHelper->getGroupId(), $upload_pk);
+      if ($upload !== true) {
+        return $response->withJson($upload->getArray(), $upload->getCode());
+      }
+      $allUploadsPerm = $reqBody['allUploadsPermission'] ? 1 : 0;
+      $newgroup = intval($reqBody['newGroupId']);
+      $newperm = intval($this->getEquivalentValueForPermission($reqBody['newPermission']));
+      $public_perm = isset($reqBody['publicPermission']) ? $this->getEquivalentValueForPermission($reqBody['publicPermission']) : -1;
+
+      $query = "SELECT perm, perm_upload_pk FROM perm_upload WHERE upload_fk=$upload_pk and group_fk=$newgroup;";
+      $result = $dbManager->getSingleRow($query, [], "");
+      if ($result) {
+        $perm_upload_pk = intVal($result['perm_upload_pk']);
+        $perm = intVal($result['perm']);
+      }
+
+      $uploadPermissionObj->editPermissionsForUpload($commu_status, $folder_pk, $upload_pk, $allUploadsPerm, $perm_upload_pk, $perm, $newgroup, $newperm, $public_perm);
+
+      $returnVal = new Info(200, "Permissions updated successfully!", InfoType::INFO);
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    } else {
+      $returnVal = new Info(409, "Scheduler is not running!", InfoType::ERROR);
```

Comment:
As per the yaml file
```suggestion
      $returnVal = new Info(503, "Scheduler is not running!", InfoType::ERROR);
```